libndt7/src/main/java/io/ooni/mk/nuvolari/libndt7/Client.java

```diff
@@ -1,100 +1,155 @@
 package io.ooni.mk.nuvolari.libndt7;
 
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.security.cert.X509Certificate;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.TimeUnit;
 
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
-import okhttp3.WebSocketListener;
-import okhttp3.WebSocket;
 import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
 import okio.ByteString;
 
-import org.json.JSONException;
-import org.json.JSONObject;
-
 // TODO(bassosimone): do we need locking for this class?
 
 public class Client extends WebSocketListener {
-    public Client(Settings settings) {
+
+    @NonNull
+    private final Settings settings;
+
+    private double count = 0.0;
+    private boolean rv = true;
+    private long t0 = 0;
+    private long tLast = 0;
+    private final long measurementInterval = TimeUnit.NANOSECONDS.convert(250, TimeUnit.MILLISECONDS);
+
+    @NonNull
+    private static final String TAG = "Client";
+
+    public Client(@NonNull Settings settings) {
```

Comment:
So, `@NonNull` will throw if passed a `null` pointer right?

Reply:
no those don't throw.  Kotlin will throw if you violate it's nullability contracts.  These are just for compiler to warn the programmer.